File: api/src/main/java/api/dtos/requests/UserRegistrationRequestDTO.java
package api.dtos.requests;

import api.models.User;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record UserRegistrationRequestDTO(
        @NotBlank(message = "Informe o seu nome.")
        @Size(min = 5, max = 60, message = "O nome deve conter entre {min} e {max} caracteres.")
        String name,
        @NotBlank(message = "Informe o seu email.")
        @Size(min = 5, max = 80, message = "O email deve conter entre {min} e {max} caracteres.")
        @Email(message = "O email informado é inválido.")
        String email,
        @NotBlank(message = "Informe a sua senha.")
        @Size(min = 6, message = "A senha deve conter no mínimo {min} caracteres.")
        String password
) {
    public User convert() {
        return User
                .builder()
                .name(name())
                .email(email())
                .password(password())
                .build();
    }
}

File: web/src/core/dtos/requests/user-registration-request.dto.ts
import { z } from 'zod';

interface UserRegistrationRequestDTO {
  name: string;
  email: string;
  password: string;
}

const validationSchema = z.object({
  name: z.string()
    .min(5, { message: "O nome deve conter no mínimo 5 caracteres." })
    .max(60, { message: "O nome deve conter no máximo 60 caracteres." }),
  email: z.string()
    .email({ message: "O email informado é inválido." })
    .min(5, { message: "O email deve conter no mínimo 5 caracteres." })
    .max(80, { message: "O email deve conter no máximo 80 caracteres." }),
  password: z.string()
    .min(6, { message: "A senha deve conter no mínimo 6 caracteres." })
});

export default UserRegistrationRequestDTO;
export { validationSchema }
